core-bundle/src/Routing/LegacyRouteProvider.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing;
+
+use Symfony\Cmf\Component\Routing\RouteProviderInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Route;
+
+class LegacyRouteProvider implements RouteProviderInterface
+{
+    /**
+     * @var FrontendLoader
+     */
+    private $frontendLoader;
+
+    /**
+     * @var RouteProviderInterface
+     */
+    private $routeProvider;
+
+    public function __construct(FrontendLoader $frontendLoader, RouteProviderInterface $routeProvider)
+    {
+        $this->frontendLoader = $frontendLoader;
+        $this->routeProvider = $routeProvider;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRouteCollectionForRequest(Request $request)
+    {
+        return $this->routeProvider->getRouteCollectionForRequest($request);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRouteByName($name)
+    {
+        if ('contao_frontend' === $name || 'contao_index' === $name) {
```

Comment:
Why can we no longer have these routes in the `routing.yml` file?

Reply:
They must not be matched by the routing, but they must be available to the URL generator.